homeassistant/components/motion_blinds/button.py

```diff
@@ -0,0 +1,66 @@
+"""Support for Motionblinds button entity using their WLAN API."""
+
+from __future__ import annotations
+
+from motionblinds.motion_blinds import LimitStatus
+
+from homeassistant.components.button import ButtonEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EntityCategory
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN, KEY_COORDINATOR, KEY_GATEWAY
+from .entity import MotionCoordinatorEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Perform the setup for Motionblinds."""
+    entities: list[ButtonEntity] = []
+    motion_gateway = hass.data[DOMAIN][config_entry.entry_id][KEY_GATEWAY]
+    coordinator = hass.data[DOMAIN][config_entry.entry_id][KEY_COORDINATOR]
+
+    for blind in motion_gateway.device_list.values():
+        if blind.limit_status == LimitStatus.Limit3Detected.name:
+            entities.append(MotionGoFavoriteButton(coordinator, blind))
+            entities.append(MotionSetFavoriteButton(coordinator, blind))
+
+    async_add_entities(entities)
+
+
+class MotionGoFavoriteButton(MotionCoordinatorEntity, ButtonEntity):
+    """Button entity to go to the favorite position of a blind."""
+
+    _attr_translation_key = "go_favorite"
+
+    def __init__(self, coordinator, blind):
+        """Initialize the Motion Button."""
+        super().__init__(coordinator, blind)
+        self._attr_unique_id = f"{blind.mac}-go-favorite"
```

Comment:
```suggestion
        self._attr_unique_id = f"{blind.mac}_go-favorite"
```
Not sure what you use at the other platforms, if it's a `-`, just keep it like that, otherwise I'd recommend this as it's a good practice because if you ever have to migrate in the future you can just do `.split("_")` and have both sides intact

Reply:
In the other platforms I also already use `self._attr_unique_id = f"{blind.mac}-battery"` in multiple places in a simular way.

Although a nice suggestion, i don't think it is wise to deviate from the pattern for the motion_blinds integrations.
Will keep this in mind for future integrations.